lte/gateway/c/session_manager/test/test_session_state_5g.cpp

```diff
@@ -0,0 +1,319 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include <future>
+#include <memory>
+#include <utility>
+
+#include "Consts.h"
+#include "magma_logging.h"
+#include "ProtobufCreators.h"
+#include "SessiondMocks.h"
+#include "SessionState.h"
+#include "SessionStateTester5g.h"
+
+using ::testing::Test;
+
+namespace magma {
+
+TEST_F(SessionStateTest5G, test_session_rules) {
```

Comment:
Please review if these test are now really required. This test is exactly the same as [test_session_rules](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Auri200%2Fmagma.git&branch=master&file=lte%2Fgateway%2Fc%2Fsession_manager%2Ftest%2Ftest_session_state.cpp&editor=JetBrains&version=v1.2.1&utm_product_name=CLion&utm_product_version=2021.1.2&start_row=30&start_col=35&end_row=30&end_col=35) in 4g. 

Also, in case this test is hitting 5g path, it is not asserting if the rule was processed as a 4g or 5g. You will have to check specific different fields (if any) to make sure.

Please do the rest of the other tests

Reply:
Also activate_rules are the same as 4g case. If a 4g rule looks like the same as a 5g rule inside sessiond, there is no need to define again the function.

Reply:
The "test_session_rules" is not required. I have removed the TC.
I have added the 5g conditions in all TCs like as below:
+  // For 5g verification
+  EXPECT_TRUE(session_state->is_5g_session());

+  // For 5g verification
+  EXPECT_EQ(0, pending_activation[0].teids.agw_teid());
+  EXPECT_EQ(0, pending_activation[0].teids.enb_teid());

+  // For 5g verification
+  EXPECT_EQ(0, pending_bearer_setup.size());

Reply:
Yes, the definition of activate_rules is same as 4g case. but it is defined into SessionStateTest class for 4g. We are using SessionStateTest5G class for 5g because build_sm_context and SessionState are different from 4g.

Reply:
If we want to move this function from SessionStateTest to SessionState then will use for both.

Reply:
Hey @prabinakpattnaik let's not add a test function into production code. Test functions should stay on test  files.

Since it is a test helper, lets leave the functions as they are even it is repeated code